notebooks/analysis/art_of_science_query.ipynb

```diff
@@ -0,0 +1,195 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "\n",
+    "with open(\"passwords.json\", \"r\") as f: \n",
+    "    sensitive_data = json.load(f)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import weco_datascience\n",
+    "from weco_datascience.reporting import get_recent_data\n",
+    "\n",
+    "df = get_recent_data(config=sensitive_data, n=100000, index=\"conversion*\")\n",
+    "df.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Grab searches"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "searches=df.loc[(df['page.name']=='images') | (df['page.name']=='works') & (df['page.query.query'].notnull())\n",
+    "                & (df['properties.totalResults'].notnull()) & (df['page.query.page']==1) | \n",
+    "                (df['page.query.page'].notnull())]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Which anonymousIds viewed page.query.id:fuc864rh?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "viewed=df.loc[(df['page.query.workId']=='fuc864rh')]\n",
+    "keepers=['anonymousId']\n",
+    "viewed=viewed[keepers]\n",
+    "\n",
+    "viewed.nunique()\n",
+    "#there were 39 anonymousIds"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "searched=['page.query.query', 'anonymousId']\n",
+    "searches2=searches[searched]\n",
+    "searches2.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Grab the searches by the anonymousIds that viewed page.query.id:fuc864rh."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "result=viewed.merge(searches2, how='left', on='anonymousId')"
```

Comment:
merging pandas dataframes is a notoriously treacherous operation, and running it with the changing code from earlier☝️, I'm seeing all sorts of weird results. I usually try to avoid merging dataframes whenever possible.

I don't think the funky merge is necessary if we just take advantage of the original `search` dataframe! The following feels a bit more readable and reliable to me:

```python
views = df[df['page.query.workId']=='fuc864rh']

unique_viewers = views['anonymousId'].unique()

result = searches[searches.anonymousId.isin(unique_viewers)]
```

Reply:
Thanks for looking at this. "isin" looks like it's going to be my new best friend!